tests/unit/resources/servers/test_server_profile_templates.py

```diff
@@ -49,89 +51,53 @@ def test_get_all(self, mock_get_all):
         mock_get_all.assert_called_once_with(
             start=2, count=500, filter=query_filter, sort=sort, scope_uris=scope_uris)
 
-    @mock.patch.object(ResourceClient, 'get')
-    def test_get_by_id(self, mock_get):
-        template_id = "6fee02f3-b7c7-42bd-a528-04341e16bad6"
-
-        self._resource.get(template_id)
-        mock_get.assert_called_once_with(id_or_uri=template_id)
-
-    @mock.patch.object(ResourceClient, 'get_by')
-    def test_get_by_property(self, mock_get_by):
-        template_property = "name"
-        template_name = "BL460c Gen8 1"
-
-        self._resource.get_by(template_property, template_name)
-        mock_get_by.assert_called_once_with(template_property, template_name)
-
-    @mock.patch.object(ResourceClient, 'get_by_name')
-    def test_get_by_name(self, mock_get_by_name):
-        template_name = "BL460c Gen8 1"
-
-        self._resource.get_by_name(template_name)
-        mock_get_by_name.assert_called_once_with(template_name)
-
-    @mock.patch.object(ResourceClient, 'create')
+    @mock.patch.object(ResourceHelper, 'create')
     def test_create(self, mock_create):
-        uri = "/rest/server-profile-templates?force=True"
         template = dict(name="BL460c Gen8 1")
 
-        self._resource.create(resource=template, timeout=TIMEOUT)
+        self._resource.create(template, timeout=TIMEOUT)
         mock_create.assert_called_once_with(
-            resource=template,
-            uri=uri,
-            timeout=TIMEOUT,
-            default_values=ServerProfileTemplate.DEFAULT_VALUES
+            template, None, -1, force=True
         )
 
-    @mock.patch.object(ResourceClient, 'update')
-    def test_update(self, mock_update):
-        uri = "/rest/server-profile-templates/4ff2327f-7638-4b66-ad9d-283d4940a4ae"
+    @mock.patch.object(Resource, 'ensure_resource_data')
+    @mock.patch.object(ResourceHelper, 'update')
+    def test_update(self, mock_update, mock_ensure_client):
         template = dict(name="BL460c Gen8 1", macType="Virtual")
 
-        self._resource.update(resource=template, id_or_uri=uri)
-        mock_update.assert_called_once_with(
-            resource=template,
-            uri=uri,
-            default_values=ServerProfileTemplate.DEFAULT_VALUES,
-            force=True
-        )
+        self._resource.update(template)
+        template["uri"] = self.uri
 
-    @mock.patch.object(ResourceClient, 'delete')
-    def test_delete(self, mock_delete):
-        template = dict(name="BL460c Gen8 1")
+        mock_update.assert_called_once_with(template, self.uri, True, -1)
 
-        self._resource.delete(resource=template, timeout=TIMEOUT)
-        mock_delete.assert_called_once_with(resource=template, timeout=TIMEOUT, force=False)
+    @mock.patch.object(ResourceHelper, 'delete')
+    def test_delete(self, mock_delete):
+        self._resource.delete(timeout=TIMEOUT)
```

Comment:
Just a doubt, why 'template' is not passing for delete call where as it is passing for create ? I saw similar calls for 'delete' in other PRs also.

Reply:
Create method needs data to create the resource and it returns the object of the created resource.
to delete a resource we have to access that resource object using name/URI and delete it. Eg obj.delete(query param).